File: dfs.py
#DFS


def findAllPossibles(graph, start):
  #create a empty set and list with start point
  visited, stack = set(), [start] 
  while stack:
    vertex = stack.pop() #set to top of stack
    if vertex not in visited:
      visited.add(vertex)
      #subtract the visited from the next node-set and 
      #push to stack all remaining nodes.
      stack.extend(graph[vertex]- visited) 
  return visited
  
  
def findPathBetween(graph, start, end):
  stack, paths = [(start, [start])], []         # use tuple to create a psuedo node (could of used class Node object instead).
  while stack:
    (vertex, path) = stack.pop() #set to top of stack
    
    # traverse only nodes not in path already
    for node in graph[vertex]- set(path):
      if node == end:
        paths.append((node, path + [node]))# found end. store for return
      else:
        stack.append((node, path+[node])) # add to stack existing path and recuse to next loop
  return paths
        

File: dfs.js
let graph = {'A': new Set(['B', 'C']),
         'B': new Set(['A', 'D', 'E']),
         'C': new Set(['A', 'F']),
         'D': new Set(['B']),
         'E': new Set(['B', 'F']),
         'F': new Set(['C', 'E'])}


function findAllPathsFrom (graph, start , end){
  let node = {
    location: start,
    data: [start]
  },stack = [node],paths = [],path, vertex;

  while(stack){ 
    vertex = stack.shift();
    path = new Set(vertex.data);
    let difference = new Set([...graph[vertex.location]].filter(x => !path.has(x))), d = Array.from(difference);
    for( let key in d){
       node = {   
          location: d[key],
          data: Array.from(path)+[d[key]],  
        };
      console.log(node)
      if (node.location === end){
          paths.push(node);
          console.log(node)
      }else{
          stack.push(node);
      }
        
    }
        
  }
  return paths;
    
}
